bitcoin/src/address.rs

```diff
@@ -727,34 +852,34 @@ impl Address {
         }
     }
 
-    /// Returns true if the given pubkey is directly related to the address payload.
+    /// Checks whether network of this address is as expected.
     ///
-    /// This is determined by directly comparing the address payload with either the
-    /// hash of the given public key or the segwit redeem hash generated from the
-    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
-    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
-        let pubkey_hash = pubkey.pubkey_hash();
-        let payload = self.payload.as_bytes();
-        let xonly_pubkey = XOnlyPublicKey::from(pubkey.inner);
-
-        (*pubkey_hash.as_ref() == *payload)
-            || (xonly_pubkey.serialize() == *payload)
-            || (*segwit_redeem_hash(&pubkey_hash).as_ref() == *payload)
+    /// For details about this mechanism, see section [*Parsing addresses*](Address#parsing-addresses)
+    /// on [`Address`].
+    #[inline]
+    pub fn expect_network(self, expected: Network) -> Result<Address, Error> {
```

Comment:
Sorry about late bikeshedding but since `.expect()` panics this may give impression it panics too (in the code, not in the doc). Maybe there's a better name? `require_network`?

Reply:
I think that is an absolutely valid concern. I changed it to `require_network`, it's a good name, too.